### ISSUE TITLE
Shortcuts

### ISSUE THREAD

**Issue by DanPristupov** [ForkIssues/Tracker#309]
I think we need to list all the shortcuts in one place.

##### Commit View
<kbd>⌘</kbd><kbd>⇧</kbd><kbd>A</kbd> - Toggle amend mode

**Comment by djbe** [comment]
Commit view: `cmd` + `enter` triggers the "Commit x Files" button

**Comment by KentVu** [comment]
Not sure if this was reported but:
the `⌘⇧B - New branch` command should create branch at the highlighted commit.
It's currently always create a new branch at the same commit of the checked out branch.
Please check!

**Edit**: OK this was reported :') #128

**Comment by SassNinja** [comment]
Something I'd really appreciate is an option to override all those shortcuts so you can customize git-fork the way you can work best.

**Comment by JessyCatterwaul** [comment]
The shortcut for Open is horrible, and I use that function a lot. Can we get a better one? In SourceTree, I typically double-click because I'm probably mousing when using this feature. Considering that's taken, in Fork, a modified click is really what I'd prefer.

**Comment by DevilDimon** [comment]
Any way to add ⌘+. to stop all current operations (as in Xcode, Terminal and elsewhere in the OS)?

**Comment by pmlandwehr** [comment]
I'd like to put in a bid for `j` and `k` to navigate up and down items in the active pane.  Much easier than up-arrow and down-arrow on a laptop keyboard, at least for me.

**Comment by infotexture** [comment]
> Something I'd really appreciate is an option to override all those shortcuts

@SassNinja On macOS, you can customize shortcuts for any command that appears in a menu via **System Preferences > Keyboard > Shortcuts > App Shortcuts** — so as long as it's in a menu somewhere, you can change it yourself.

**Comment by Tricertops** [comment]
@infotexture The issue is that not all of them are in the main menu, especially not those that operate on currently selected commit or file.

I already customized this way quick actions: Quick Push ⌘P, Quick Pull ⌘L, Quick Stash ⌘H, and their non-quick variants using ⌥ key. However, this customization only works if:
1. The action is in the main menu, even if hidden (some Fork actions are not).
2. The shortcut must use ⌘ key (so can’t override Space, Return, or even double-click).

Regarding the above points, Fork should:
1. Expose commit and file actions in the main menu.
2. Allow customizing the non-⌘ actions in the preferences.

cc @DanPristupov

**Comment by Tricertops** [comment]
@DanPristupov You should mention that <kbd>Space</kbd> opens side-by-side diff of selected file.

The shortcut for opening a selected file should really be more accessible than opening a new repo. I would suggest to have Open File as ⌘O and Open Repo as ⇧⌘O.

As a long-time SourceTree user, I’m quite used to double-click to open file and <kbd>Space</kbd> to stage/unstage a file. I’m trying to get used to the ⌘S, since I didn’t know about the <kbd>Return</kbd> key. However, opening a file using ⌥⇧⌘O is too much 😆

**Comment by tkainrad** [comment]
KeyCombiner now has a table of all Fork keyboard shortcuts that are listed in this issue and its Windows counterpart: https://keycombiner.com/collections/fork/macos/

It can be searched and filtered.

While creating this resource, I noticed that the macOS version of Fork does not have the _Zoom in/out_, _Filter_, _Rename Repository_, and _Remove Repository_ shortcuts that the Windows version has. Is this correct or is it just missing in this issue?

![image](https://user-images.githubusercontent.com/7394822/111536091-1a275300-876a-11eb-807c-5f955086dc79.png)

**Comment by konclave** [comment]
It would be nice to have a shortcut that set focus to Unstaged and Staged files lists. Because if you get to the "uncommited changes" view with the Cmd-1 shortcut, you can get to the Staged files list only with the mouse click. By pressing "tab" key it focuses only one of the lists that has focus the last and skips another.
E.g.: on All changes press Cmd-1. You're moved to Uncommeted changes with the focus set to Unstaged list. Press Cmd-1. Focus is the "Commit subject" field. Now I can move back to the Unstaged list with the tab key, but Staged list is unreachable. 
If I focus Staged list with the mouse, then I can move with the tab to the "Commit subject" and back, but Unstaged list is unreachable.

**Comment by DanPristupov** [comment]
> Now I can move back to the Unstaged list with the tab key, but Staged list is unreachable.

Tab switches unstaged -> staged -> commit subject -> description. Shift+Tab switches backward. The focus might not be visible when there's no item selected, but it's there.

**Comment by konclave** [comment]
> Tab switches unstaged -> staged -> commit subject -> description. Shift+Tab switches backward. The focus might not be visible when there's no item selected, but it's there.
Thanks. It's really there, just need to select some file with the arrows.

**Comment by AndrewSouthpaw** [comment]
[UPDATE] Nevermind seems to have been an ephemeral problem, restarting Fork fixed the issue.

So, it _used_ to do this cycle for Tabs...

https://i.imgur.com/AEAIewP.jpg

But recently that changed, and now it goes to the toolbar for the diff window:

https://i.imgur.com/6aOt5vi.jpg

I'd love to see the old cycle come back, because it was much more ergonomic. Now I can't stay keyboard only with Fork. 😞

**Comment by konclave** [comment]
Is there a hotkey to switch between Branches - Stashes - Search commits tabs?

**Comment by DanPristupov** [comment]
@konclave no, there's no keyboard shortcut for that. And since we are here, please read my comment in the other topic: https://github.com/fork-dev/Tracker/issues/1696#issuecomment-1283678681.

**Comment by NSErfan** [comment]
Hey @DanPristupov, I used to be able to do `⌘ + B` and get a search field to checkout a branch and that has stopped working as of `version 2.28`.  That's quite handy.  Could you please bring that back?

**Comment by niorko** [comment]
> Hey @DanPristupov, I used to be able to do `⌘ + B` and get a search field to checkout a branch and that has stopped working as of `version 2.28`. That's quite handy. Could you please bring that back?

I also have the same issue 😭. Please bring it back. 🙏

**Comment by DanPristupov** [comment]
Can you use `Cmd+P` -> Checkout instead?

**Comment by matthewjamesadam** [comment]
@DanPristupov I used `Cmd+B` very often as well.  Cmd+P -> Checkout -> branchname feels pretty cumbersome in comparison.

**Comment by NSErfan** [comment]
Yeah, and we still have `⌘ + shift + B` to create a new branch which is handy as well.  But I checkout branches way more often than creating new branches, so bringing back the sibling shortcut `⌘ + B` to checkout would be super useful.

**Comment by leshque** [comment]
Same here, after the update I really miss the `Cmd+B` shortcut to switch between branches.

**Comment by tomaskraina** [comment]
I have to agree what was written above. I got to this issue out of frustration that `CMD+B` stopped working and I could not find a menu item with a shortcut to do that.

Using `CMD+P` is suboptimal and degrades the UX for me quite a bit.

@DanPristupov  Could you please reconsider adding checkout branch menu item + shortcut back?

**Comment by NSErfan** [comment]
Hey @DanPristupov, thanks for bringing back `⌘+B for` checkout. I've been checking for updates almost every day, and I got really happy when I saw the update today. :)

**Comment by ConstantineLoutas** [comment]
Hey @DanPristupov, something changed recently and the `Quick Launch...` option is now invoked by both:
* `⌘` `P` (documented above, so intentional), and
* `⇧` `⌘` `O` (undocumented, probably unintentional).

If the latter was indeed unintentional, can we get it back? I was using it for something else and now `Quick Launch...` takes precedence 😅

**Comment by ottob** [comment]
If I'm in the commit list and use ⌘ F to search for something, I can not then use ⌘ G to search for the next matching item. Can you please add this?

**Comment by clementlize** [comment]
Hi there, would be great to have a shortcut to check the "amend" checkbox in the Changes View :)

**Comment by clementlize** [comment]
Also, I find the shortcut to create a tag sometimes misleading, as it creates a tag on the **checked out commit** and not the one selected (in blue)

**Comment by SebKranz** [comment]
> Hi there, would be great to have a shortcut to check the "amend" checkbox in the Changes View :)

`⌘` `⇧` `A`

**Comment by zev-levitramos-zocdoc** [comment]
@DanPristupov hey! It seems like there might be a small oddity when hitting delete to remove a highlighted branch or stash. 
It seems that if the branch filter is on for the current branch, and you highlight a different branch and press  `delete` nothing happens. It only works fine when the branch filter is off and the commit pane is able to highlight the specific commit head for the non-current branch.

Not sure if this is intentional, but can be a little confusing as to why it doesn't work.

**Comment by ian-h-chamberlain** [comment]
> The shortcut for opening a selected file should really be more accessible than opening a new repo. I would suggest to have Open File as ⌘O and Open Repo as ⇧⌘O.

@Tricertops — I know you posted this a while ago, but maybe this will help you or others who are looking for this functionality... as others have mentioned, macOS actually allows you to customize these shortcuts and I found this configuration worked for me in `System Settings > Keyboard > Keyboard Shortcuts... > App Shortcuts`:

<img width="436" alt="Screenshot 2024-06-12 at 12 26 40" src="https://github.com/fork-dev/Tracker/assets/11131775/574fa9b9-f9eb-4d79-8fd5-c3443c1ed8c4">

The first shortcut triggers the `File > Open...` dialog, and the second option triggers the context menu `Open` command that appears when you right-click/ctrl-click a file. Hope this helps if you are still looking for that funcitonality!

Edit: From Apple's [documentation](https://support.apple.com/guide/mac-help/create-keyboard-shortcuts-for-apps-mchlp2271/mac), it seems like maybe it would be less ambiguous/more correct to use `File->Open...` as the shortcut action:
<img width="432" alt="Screenshot 2024-06-12 at 12 35 49" src="https://github.com/fork-dev/Tracker/assets/11131775/ca9ee429-15a5-4ba6-a8ff-0a7372179a0b">

Both seem to work for me but specifying the `File` menu "feels better", I guess 😆

**Comment by wyatterp365** [comment]
> 

> 

> 

- [ ]

**Comment by fo-fo** [comment]
> @Tricertops — I know you posted this a while ago, but maybe this will help you or others who are looking for this functionality... as others have mentioned, macOS actually allows you to customize these shortcuts and I found this configuration worked for me in `System Settings > Keyboard > Keyboard Shortcuts... > App Shortcuts`:

thanks for this! the default shortcut (⌘⌥⇧O) for opening the selected file didn't work for me on a fresh install, but i was able to re-bind it with this tip!

**Comment by britalmeida** [comment]
Would it be possible to add `F3` and `Shift+F3` for search next/prev also on macOS?
That one I couldn't do via the system settings.

Fyi, I added these to system settings:
- Refresh: F5 (instead of Cmd+R)
- Filter: Cmd+F
These ones could be nice to add by default although it's possible to workaround.

I configure my mac to conform as much as possible to Linux and Windows or I go crazy switching between :)

**Comment by patrick-fu** [comment]
Can a shortcut be added to quickly switch between different tabs, such as option + 1, option + 2, option + 3, etc

Scenario: I want to quickly jump from the second tab to the seventh tab. Using the current Ctrl+Tab shortcut, I need to press it five times.
